crates/primitives/src/transaction/mod.rs

```diff
@@ -1726,12 +1726,32 @@ impl TryFrom<reth_rpc_types::Transaction> for TransactionSignedEcRecovered {
     fn try_from(tx: reth_rpc_types::Transaction) -> Result<Self, Self::Error> {
         let signature = tx.signature.ok_or(ConversionError::MissingSignature)?;
 
+        let transaction: Transaction = tx.try_into()?;
+
         TransactionSigned::from_transaction_and_signature(
-            tx.try_into()?,
+            transaction.clone(),
             Signature {
                 r: signature.r,
                 s: signature.s,
-                odd_y_parity: signature.y_parity.ok_or(ConversionError::MissingYParity)?.0,
+                odd_y_parity: match transaction.tx_type() {
+                    // If the transaction type is Legacy, adjust the v component of the signature
+                    // according to the Ethereum specification
+                    TxType::Legacy => {
+                        // Calculate the new v value based on the EIP-155 formula:
+                        // v = {0,1} + CHAIN_ID * 2 + 35
+                        signature.v -
+                            U256::from(if let Some(chain_id) = transaction.chain_id() {
+                                // If CHAIN_ID is available, calculate the new v value accordingly
+                                chain_id.saturating_mul(2).saturating_add(35)
+                            } else {
+                                // If CHAIN_ID is not available, set v = {0,1} + 27
+                                27
+                            })
```

Comment:
I think so following the first paragraph of the spec here: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md

- If `(nonce, gasprice, startgas, to, value, data)` then `v = {0,1} + 27 `
- If `(nonce, gasprice, startgas, to, value, data, chainid, 0, 0)` then `v = {0,1} + CHAIN_ID * 2 + 35`

Reply:
The point is that we need to instead confirm with implementations rather than just look at the spec, because the spec does not define RPC behavior

looks like geth only returns chainid if it's eip155, and we do the same (ie, we just return from db), so this should be fine
https://github.com/ethereum/go-ethereum/pull/25155/files

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/api/transactions.rs#L742-L765